backend-api/src/functions/asyncIssueBiometricCredential/validateSqsEvent.ts

```diff
@@ -0,0 +1,78 @@
+import { SQSEvent } from "aws-lambda";
+import { emptyFailure, Result, successResult } from "../utils/result";
+import { logger } from "../common/logging/logger";
+import { LogMessage } from "../common/logging/LogMessage";
+import { validateSessionId } from "../common/request/validateSessionId/validateSessionId";
+
+export const validateSqsEvent = (event: SQSEvent): Result<string, void> => {
+  if (event == null) {
+    logger.error(LogMessage.ISSUE_BIOMETRIC_CREDENTIAL_INVALID_SQS_EVENT, {
+      errorMessage: "Event is either null or undefined.",
+    });
+    return emptyFailure();
+  }
+
+  if (event.Records == null) {
+    logger.error(LogMessage.ISSUE_BIOMETRIC_CREDENTIAL_INVALID_SQS_EVENT, {
+      errorMessage: "Invalid event structure: Missing 'Records' array.",
+    });
+    return emptyFailure();
+  }
+
+  if (event.Records.length !== 1) {
+    logger.error(LogMessage.ISSUE_BIOMETRIC_CREDENTIAL_INVALID_SQS_EVENT, {
+      errorMessage: `Expected exactly one record, got ${event.Records.length}.`,
+    });
+    return emptyFailure();
+  }
+
+  const record = event.Records[0];
+  if (record.body == null) {
+    logger.error(LogMessage.ISSUE_BIOMETRIC_CREDENTIAL_INVALID_SQS_EVENT, {
+      errorMessage: "Event body either null or undefined.",
+    });
+    return emptyFailure();
+  }
+
+  let parsedBody: unknown;
+  try {
+    parsedBody = JSON.parse(record.body);
+  } catch {
+    logger.error(LogMessage.ISSUE_BIOMETRIC_CREDENTIAL_INVALID_SQS_EVENT, {
+      errorMessage: `Failed to parse event body. Body: ${record.body}`,
+    });
+    return emptyFailure();
+  }
+
+  if (!isParsedBody(parsedBody)) {
+    logger.error(LogMessage.ISSUE_BIOMETRIC_CREDENTIAL_INVALID_SQS_EVENT, {
+      errorMessage: `Parsed body not in expected shape: ${JSON.stringify(parsedBody)}`,
+    });
+    return emptyFailure();
+  }
+
+  const { sessionId } = parsedBody;
+  const validateSessionIdResult = validateSessionId(sessionId);
+  if (validateSessionIdResult.isError) {
+    logger.error(LogMessage.ISSUE_BIOMETRIC_CREDENTIAL_INVALID_SQS_EVENT, {
+      errorMessage: validateSessionIdResult.value.errorMessage,
+    });
+    return emptyFailure();
+  }
+
+  return successResult(sessionId);
+};
+
+const isParsedBody = (
+  parsedBody: unknown,
+): parsedBody is { sessionId: string } => {
```

Comment:
Note: I did have an interface for this previously but felt it was overkill so removed. If you think it would be better to re add I am happy to. It would look something like:

```suggestion
interface ParsedBody {
  sessionId: string;
}

const isParsedBody = (
  parsedBody: unknown,
): parsedBody is ParsedBody => {
```

Reply:
IMO if it's not used anywhere else we may as well have it inline like you do currently.